forms/LabelField.php

```diff
@@ -1,30 +1,38 @@
 <?php
+
 /**
- * Simple label tag. This can be used to add extra text in your forms.
- * Consider using a {@link ReadonlyField} if you need to display a label
- * AND a value.
- * 
+ * Simple label, to add extra text in your forms.
+ *
+ * Use a {@link ReadonlyField} if you need to display a label and value.
+ *
  * @package forms
  * @subpackage fields-dataless
  */
 class LabelField extends DatalessField {
-	
 	/**
 	 * @param string $name
-	 * @param string $title
-	 * @param Form $form
+	 * @param null|string $title
 	 */
-	public function __construct($name, $title) {
-		// legacy handling for old parameters: $title, $heading, ...
-		// instead of new handling: $name, $title, $heading, ...
+	public function __construct($name, $title = null) {
+		// legacy handling:
+		// $title, $headingLevel...
 		$args = func_get_args();
+
 		if(!isset($args[1])) {
 			$title = (isset($args[0])) ? $args[0] : null;
-			$name = $title;
-			$form = (isset($args[3])) ? $args[3] : null;
-		} 
-		
+
+			if(isset($args[0])) {
+				$title = $args[0];
+			}
+
+			// Prefix name to avoid collisions.
+			$name = 'LabelField' . $title;
+
+			if(isset($args[3])) {
+				$this->setForm($args[3]);
```

Comment:
form parameter in the constructor is actually deprecated (I think in 3.1.0 or 3.0.0). I found this out after merging a "fix" for a field by setting 'setForm' and making someone else angry upstream, I think in FormAction?. I think we should just drop this.